src/storage/disk_manager.c

```diff
@@ -1222,16 +1222,38 @@ int
 disk_rv_redo_dboutside_newvol (THREAD_ENTRY * thread_p, const LOG_RCV * rcv)
 {
   DISK_VOLUME_HEADER *const vhdr = (DISK_VOLUME_HEADER *) rcv->data;
-  const char *const vol_fullname_other_config = disk_vhdr_get_vol_fullname (vhdr);
 
-  char buf_vol_full_name[PATH_MAX];
-  disk_adapt_volume_full_name_to_local_config (buf_vol_full_name, vol_fullname_other_config);
+  if (is_tran_server_with_remote_storage () && is_passive_transaction_server ())
+    {
+      /* Passive transaction server with remote storage should not do fileio_format(),
+       * and it only uses disk_Page_server_perm_volume_count to check if there is a permanent volume in page server.
+       * Since PTS do not have a disk header page for newly added volume at the time of replication,
+       * disk_rv_redo_format (), which requires disk header page and does update the disk_Page_server_perm_volume_count,
+       * is not called during replication.
+       * Therefore, disk_Page_server_perm_volume_count for PTS is updated here when the new volume is created.
+       *
+       * NOTE : If it is decided to use disk_Cache in PTS, this part can be moved to disk_rv_redo_format ()
+       *        where disk_Cache is updated.
+       */
 
-  if (fileio_find_volume_descriptor_with_label (buf_vol_full_name) == NULL_VOLDES)
+      if (vhdr->type == DB_PERMANENT_VOLTYPE)
+	{
+	  disk_Page_server_perm_volume_count++;
+	}
```

Comment:
```suggestion
	  disk_Page_server_perm_volume_count++;
```
It doesn't seem that it can be DB_TEMPORARY_VOLTYPE.

Reply:
OK, Thanks

Reply:
Better assert to make intent explicit.